components/pkg-export-kubernetes/src/manifestjson.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2017 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use handlebars::Handlebars;
+use serde_json::Value;
+
+use export_docker::Result;
+
+use manifest::Manifest;
+use error::Error;
+
+// Kubernetes manifest template
+const MANIFESTFILE: &'static str = include_str!("../defaults/KubernetesManifest.hbs");
+const BINDFILE: &'static str = include_str!("../defaults/KubernetesBind.hbs");
+
+pub struct ManifestJson {
+    pub main: Value,
+    pub binds: Vec<Value>,
+}
+
+impl ManifestJson {
+    pub fn new(manifest: &Manifest) -> Result<Self> {
```

Comment:
Is there any consistency in habitat or in rust stdlib about `new` functions returning `Result<Self>`? I thought that `new` basically should return `Self`, so if a constructor function returns `Result` then maybe it should be named differently?

Reply:
This would be moot if Manifest could guarantee the validity of binds, because checking the bind string looks like the only reason why we return `Result` here.

Reply:
returning a Result is pretty norm AFAIK. I've seen it being done. It's like _new() functions in C returning a NULL and a GError for me.